From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Laura Abbott <labbott@redhat.com>
Date: Thu, 2 Jan 2020 14:24:45 -0500
Subject: [PATCH] mptspi: pci-id table changes

Author: Tomas Henzl <thenzl@redhat.com>
Date:   Thu Jan 10 16:11:44 2019 -0500

    [message] mptspi: pci-id table changes

    Message-id: <20190110161145.5117-3-thenzl@redhat.com>
    Patchwork-id: 237190
    O-Subject: [RHEL-8.0 e-stor 2/3] mptspi: pci-id table changes
    Bugzilla: 1651803
    RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
    RH-Acked-by: Jarod Wilson <jarod@redhat.com>
    RH-Acked-by: Chris Leech <cleech@redhat.com>

    RHEL-only
    We want to provide this driver only for the virtualisation use case,
    so we remove all other entries from the pci-id table and add only the
    single entry needed for virtualisation.

    Signed-off-by: Herton R. Krzesinski <herton@redhat.com>

Upstream Status: RHEL only
Signed-off-by: Laura Abbott <labbott@redhat.com>
---
 drivers/message/fusion/mptspi.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/message/fusion/mptspi.c b/drivers/message/fusion/mptspi.c
index c028f5944694..1f458e35effb 100644
--- a/drivers/message/fusion/mptspi.c
+++ b/drivers/message/fusion/mptspi.c
@@ -1238,12 +1238,17 @@ static struct spi_function_template mptspi_transport_functions = {
  */

 static struct pci_device_id mptspi_pci_table[] = {
+#ifdef CONFIG_RHEL_DIFFERENCES
+	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_53C1030,
+	PCI_VENDOR_ID_VMWARE, PCI_ANY_ID },
+#else
 	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_53C1030,
 		PCI_ANY_ID, PCI_ANY_ID },
 	{ PCI_VENDOR_ID_ATTO, MPI_MANUFACTPAGE_DEVID_53C1030,
 		PCI_ANY_ID, PCI_ANY_ID },
 	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_53C1035,
 		PCI_ANY_ID, PCI_ANY_ID },
+#endif
 	{0}	/* Terminating entry */
 };
 MODULE_DEVICE_TABLE(pci, mptspi_pci_table);
-- 
2.26.2

